Update units-of-measure.md (#37468)
dotnet/docs: docs/fsharp/language-reference/units-of-measure.md

Floating point and signed integer values in F# can have associated units of measure, which are typically used to indicate length, volume, mass, and so on. By using quantities with units, you enable the compiler to verify that arithmetic relationships have the correct units, which helps prevent programming errors.

> > [!NOTE] despite the units of measure main raison d'être is correctness in arithmetic computations, the feature can also be leveraged for adding type safe annotation with zero representation costs to other types, with approach such as [FSharp.UMX](https://github.com/fsprojects/FSharp.UMX) project.

Reviewer: Making a few comments here:

First, the NOTE tag needs to be on its own line for the build system to render it correctly.
Second, we avoid non-English phrases in our docs. Among other issues, it wreaks havoc on localization to other languages.

I made some suggestions. Feel free to accept them, or offer your own alternative wording:

```suggestion
> [!NOTE]
> These examples demonstrates correctness in arithmetic computations involving units of measure, the feature can also be leveraged for adding type safe annotation with zero representation costs to other types, with approach such as [FSharp.UMX](https://github.com/fsprojects/FSharp.UMX) project.
```

Author: @BillWagner thanks a bunch for blending my submission to make it fit both the writing style and display adjustments required to make it fit nicely in the F# documentation.

The adjustments looks great to me.
